### ISSUE TITLE
"Go to parent/child" navigation in commit graph

### ISSUE THREAD

**Issue by zhennTil** [jesseduffield/lazygit#3974]
**Is your feature request related to a problem? Please describe.**

Before switching to lazygit, I used a GUI that had a keybinding to go to the commit's first parent (usually the merge parent, in the case of a merge commit).

I find myself missing this feature when looking at the full commit graph for context, but wanting to follow a specific branch.
Also, the inverse operation of navigating to the last(?) child.

**Describe the solution you'd like**

I would like to have keybinds that act in the manner described above, navigating to the commit's parent or child.

Proof-of-concept implementation of the go-to-parent functionality here: https://github.com/zhennTil/lazygit/tree/feature/goto-parent -- demo below.

It still needs some robustness, and the go-to-child part which is a bit more difficult, but I will gladly finish this and open a PR, if there is interest.

**Describe alternatives you've considered**

Perhaps this is possible with custom commands, but I could not find a way to trigger GUI actions from custom commands -- maybe I simply missed that, or it is not currently possible.

**Additional context**

![gtp](https://github.com/user-attachments/assets/55828705-60c2-41e0-90f2-ddc64d81c6a2)

**Comment by Saafo** [comment]
Want the similar feature too! Only need to be focused on the parent commits sometimes. Maybe add an shortcut to toggle git log mode between `--first-parent` and nope just like tower or sourcetree.

**Comment by stefanhaller** [comment]
Oh yes, this would be very useful. I want to be able to toggle between `--first-parent` log and full log, and also be able to select a merge commit to fold/unfold it. Sublime Merge has this, and it's pretty awesome. (If we had that, first-parent mode could even become the default, maybe.)

Unfortunately, this would require massive changes, so it's not something we can tack on easily.

**Comment by Saafo** [comment]
Can we take it apart? First

`toggle between --first-parent log and full log`


then

`be able to select a merge commit to fold/unfold it.`

Seems less changes with only first part?

**Comment by stefanhaller** [comment]
Yes, this could help a bit, but it's still far from trivial I'm afraid.

**Comment by liontiger23** [comment]
For me navigation on full `--all` graph without filtering or folding would be much more useful personally.
I had similar request in #3575, which probably can be closed as duplicate now.
